test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java

```diff
@@ -80,19 +83,33 @@ public ClientYamlTestResponse callApi(String apiName, Map<String, String> params
         //divide params between ones that go within query string and ones that go within path
         Map<String, String> pathParts = new HashMap<>();
         Map<String, String> queryStringParams = new HashMap<>();
+
+        List<String> apiPathParts = restApi.getPathParts().entrySet().stream().filter(e -> e.getValue() == true).map(Entry::getKey)
```

Comment:
Maybe `Set<String>` instead? The list should be short so it likely doesn't matter but it feels better to me if it is a set.